### DISCUSSION TITLE
Having problem understanding how to deal with arbitrary precision

### DISCUSSION THREAD

**Discussion by StErMi** [dinerojs/dinero.js#652] [General]
Hi there,

I'm new to the library, but it seems the must-have library when dealing with currencies. I'm building a web app that deals with cryptocurrencies and their prices.

Each crypto has their own decimals and when I need to display the value of an amount of token X I need to multiply the amount for the value in dollars.

Now I'm having difficulties to understand how this should be handled via dinero. 
For example:

- balance `4564660073892638156894` (this is the balance of the user with 18 decimals)
- balance formatted 4564.660073892638156894 (formatted balance using [etherjs formatUtils](https://docs.ethers.io/v5/api/utils/display-logic/#utils-formatUnits) `formatUnits(BigNumber.from(balance), decimals)`
- Token value in USD `0.223703` (each token is worth that value, 1 token === 1e18)
- Balance * Value `1021.1281525100048`

How should I use `dinero` to correctly format this and then use it when I need to do math with different token prices to

**Comment by sarahdayan** [comment]
Hey @StErMi,

You can create Dinero objects in ETH, then convert them in USD with the [`convert`](https://v2.dinerojs.com/docs/api/conversions/convert) function and format in with [`toFormat`](https://v2.dinerojs.com/docs/api/formatting/to-format), like in this [sandbox](https://stackblitz.com/edit/node-hmdyjg?file=index.mjs).

Note that I'm using native `bigint` to safely manipulate values, but for now the formatting part can give slight imprecisions (see https://github.com/dinerojs/dinero.js/issues/294).

If you want to use Ethers formatting utilities, you'll need to create a custom calculator using `BigNumber` from the library, so you can then make your own formatting utility by wrapping [`toSnapshot`](https://v2.dinerojs.com/docs/api/formatting/to-snapshot) (until the issue linked above is fixed by https://github.com/dinerojs/dinero.js/pull/309).

For documentation on how to use a custom type and implement a calculator, please refer to [the docs](https://v2.dinerojs.com/docs/guides/using-different-amount-types#using-dinero-with-a-custom-amount-type).
